exporter/elasticsearchexporter/config.go

```diff
@@ -0,0 +1,228 @@
+	Authentication AuthenticationSettings `mapstructure:",squash"`
+
+	// ReadBufferSize for HTTP client. See http.Transport.ReadBufferSize.
+	ReadBufferSize int `mapstructure:"read_buffer_size"`
+
+	// WriteBufferSize for HTTP client. See http.Transport.WriteBufferSize.
+	WriteBufferSize int `mapstructure:"write_buffer_size"`
+
+	// Timeout configures the HTTP request timeout.
+	Timeout time.Duration `mapstructure:"timeout"`
+
+	// Headers allows users to configure optional HTTP headers that
+	// will be send with each HTTP request.
+	Headers map[string]string `mapstructure:"headers,omitempty"`
+
+	configtls.TLSClientSetting `mapstructure:",squash"`
+}
+
+// AuthenticationSettings defines user authentication related settings.
+type AuthenticationSettings struct {
+	// User is used to configure HTTP Basic Authentication.
+	User string `mapstructure:"user"`
+
+	// Password is used to configure HTTP Basic Authentication.
+	Password string `mapstructure:"password"`
+
+	// APIKey is used to configure ApiKey based Authentication.
+	//
+	// https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-create-api-key.html
+	APIKey string `mapstructure:"api_key"`
+}
+
+// DiscoverySettings defines Elasticsearch node discovery related settings.
+// The exporter will check Elasticsearch regularly for available nodes
+// and updates the list of hosts if discovery is enabled. Newly discovered
+// nodes will automatically be used for load balancing.
+//
+// DiscoverySettings should not be enabled when operating Elasticsearch behind a proxy
+// or load balancer.
+//
+// https://www.elastic.co/blog/elasticsearch-sniffing-best-practices-what-when-why-how
+type DiscoverySettings struct {
+	// OnStart, if set, instructs the exporter to look for available Elasticsearch
+	// nodes the first time the exporter connects to the cluster.
+	OnStart bool `mapstructure:"on_start"`
+
+	// Interval instructs the exporter to renew the list of Elasticsearch URLs
+	// with the given interval. URLs will not be updated if Interval is <=0.
+	Interval time.Duration `mapstructure:"interval"`
+}
+
+// FlushSettings  defines settings for configuring the write buffer flushing
+// policy in the Elasticsearch exporter. The exporter sends a bulk request with
+// all events already serialized into the send-buffer.
+type FlushSettings struct {
+	// Bytes sets the send buffer flushing limit.
+	Bytes int `mapstructure:"bytes"`
+
+	// Interval configures the max age of a document in the send buffer.
+	Interval time.Duration `mapstructure:"interval"`
+}
+
+// RetrySettings defines settings for the HTTP request retries in the Elasticsearch exporter.
+// Failed sends are retried with exponential backoff.
+type RetrySettings struct {
+	// Enabled allows users to disable retry without having to comment out all settings.
+	Enabled bool `mapstructure:"enabled"`
+
+	// Max configures how often an HTTP request is retried before it is assumed to be failed.
+	Max int `mapstructure:"max"`
```

Comment:
`MaxRequests`? If I were to add this to https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/queued_retry.go#L58 that would probably be a more consistent name.

Reply:
+1

Updated field name and configuration name to `max_requests`.